hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/BucketAssignFunction.java

```diff
@@ -159,8 +164,11 @@ public void processElement(I value, Context ctx, Collector<O> out) throws Except
     final BucketInfo bucketInfo;
     final HoodieRecordLocation location;
 
-    if (!partitionLoadState.contains(hoodieKey.getPartitionPath())) {
+    if (bootstrapIndex && !partitionLoadState.contains(hoodieKey.getPartitionPath())) {
       // If the partition records are never loaded, load the records first.
+
+      // The dataset may be huge, thus the processing would block for long,
```

Comment:
Is it the right place to add this comment?

Reply:
I think it is okey, for better understand the code logic.

Reply:
I did not question the necessity of this comment, just to see if it has a more suitable position. Because when I see this comment, the method is already executing. And your comment contains "disabled by default", but this method does not accept a flag. Personally, I think a better position is placed before the above if statement, where the execution switch is.